toolchains/vivado.py

```diff
@@ -329,3 +331,15 @@ def add_runtimes(self):
 
     def versions(self):
         return {'yosys': self.yosys_ver(), 'vivado': self.vivado_ver()}
+
+
+class VivadoYosysUhdm(VivadoYosys):
+    '''Vivado PnR + Yosys synthesis using uhdm frontend'''
+    def __init__(self, rootdir):
+        VivadoYosys.__init__(self, rootdir)
+        self.synthtool = 'yosys'
+        self.synthoptions = [
+            '-iopad', '-arch xc7', '-flatten', 'frontend=surelog'
+        ]
+        self.toolchain = 'yosys-vivado-uhdm'
+        self.library_files = f"{self.rootdir}/env/conda/envs/symbiflow-env/share/uhdm-yosys/xilinx/cells_sim.v,{self.rootdir}/env/conda/envs/symbiflow-env/share/uhdm-yosys/xilinx/cells_xtra_surelog.v"
```

Comment:
Is there a way to not have to write this explicitly? Mainly because this requires accessing a specific conda environment. Maybe you could get the share directory with `yosys-config --datdir`

Reply:
I cant use ``yosys-config`` as it returns path to datdir where this package was built, e.g.: ``/home/runner/work/conda-eda...``.

I've changed this, to only assume, that ``uhdm-yosys`` is installed in symbiflow-env, but if uhdm-yosys is in PATH, we are pointing to the library files from version in the PATH.